pkg/mads/v1/service/components.go

```diff
@@ -87,6 +88,7 @@ func NewXdsContext(log logr.Logger) (envoy_cache.NodeHash, util_xds_v3.SnapshotC
 type hasher struct{}
 
 func (_ hasher) ID(node *envoy_core.Node) string {
-	// in the very first implementation, we don't differentiate clients
-	return ""
+	// now that we start differentiating between clients are we ok with this config growing for old mechanism (under `mesh.metrics`)
```

Comment:
this needs to be addressed, I think it's fine, snapshots have cleanup mechanism and I don't think there's going to be 1000s of prometheus instances